rest_framework/pagination.py

```diff
@@ -146,6 +147,17 @@ def invert(x):
 class BasePagination(object):
     display_page_controls = False
 
+    def __init__(self, *args, **kwargs):
+        # Use of default page size setting requires a default Paginator class
+        if api_settings.PAGE_SIZE and not api_settings.DEFAULT_PAGINATION_CLASS:
+            warnings.warn(
+                "A valid paginator class must be specified with `DEFAULT_PAGINATION_CLASS` "
+                "when using the `PAGE_SIZE` default pagination setting."
+                "Defaulting the setting to specifies `PageNumberPagination` "
+                "is deprecated as pagination is  disabled by default.",
+                DeprecationWarning
+            )
```

Comment:
Is this correct? 

By setting `'DEFAULT_PAGINATION_CLASS': None` is it not the case that "Defaulting the setting to specifies `PageNumberPagination` " is simply **removed**, rather than Deprecated? 

Also, could I not set `PAGE_SIZE` globally and then just enable pagination per-view?

Reply:
I am not sure I understand your question ... this check is saying that you cannot have set a default PAGE_SIZE setting without also setting DEFAULT_PAGINATION_CLASS.  I believe the language of the second sentence could be improved but its trying to say that defaulting the variable DEFAULT_PAGINATION_CLASS to `PageNumberPagination` was removed.

The issue here is that the current default in 3.6 is that there is no page size set and PageNumberPagination doesn't have a pagination class defined, so you end up with a case where parts of the code thinks it is paginated but no pagination ever actually occurs .  The point of this warning is to prevent users that rely on the current default of PageNumberPagination where it actually works for them because they set PAGE_SIZE and during their upgrade they would realize they need to specify both settings values in their project rather than rely on this former default.   When it defaults to None it would disable pagination for those users otherwise.

More information here:  https://github.com/encode/django-rest-framework/issues/5168

Reply:
Also the only thing in DRF that uses settings.PAGE_SIZE is the PageNumberPagination class ... however it defaults to the settings PAGE_SIZE and has `page_size_query_param = None` so there is no way to change pagination by default without either setting PAGE_SIZE setting or sub-classing  PageNumberPagination.  

The idea here is that going forward you would want to be explicit in setting a per-class Paginator that makes sense, or to use both defaults to enable pagination across the project space.   I recall at Pycon that Tom wanted such a DeprecationWarning to ensure users that upgrade don't go from having pagination on in their project to disabled without any explanation.  This concern would be specifically users that set a PAGE_SIZE setting but were relying on the default DEFAULT_PAGINATION_CLASS value.

Reply:
I think you bring up a good point about users that want to "set PAGE_SIZE globally and then just enable pagination per-view" -- In this case they would specifically be using PageNumberPagination in specific views and have a global PAGE_SIZE set.   One option is to sub-class PageNumberPagination  and set the page_size setting once on that class, then use that in the per-view instances.  Open to refinement, but just stating the history of this PR.

Reply:
@matteius Thanks for the good replies. 

> The point of this warning is to prevent users that rely on the current default of PageNumberPagination where it actually works for them because they set PAGE_SIZE and during their upgrade they would realize they need to specify both settings values in their project rather than rely on this former default.

... and ...

> ...I recall at Pycon that Tom wanted such a DeprecationWarning to ensure users that upgrade don't go from having pagination on in their project to disabled without any explanation.

OK. That's the key. 

My thought here is that we should remove the Deprecation warning but add a [System Check](https://docs.djangoproject.com/en/1.11/topics/checks/#module-django.core.checks) for the same thing. (It runs once at start up and users can silence this if they really do want just `PAGE_SIZE`.) 

We can then call out the change in the Release Notes and Announcement for 3.7 and the combination of the two would be good enough. 

Does that sound OK to you? (Are you happy to write the system check?)

Reply:
Yes, this sounds like the right direction to head in.   I just reviewed the system check docs and I feel comfortable converting this PR to that approach, but where do you suggest is the right file to implement the check method and register it?  I was considering doing it in the compat.py file unless you have an alternate suggestion @carltongibson ?
Thanks!

Reply:
@matteius That sounds fine. (We can always move it later if needed.)

Thanks for the great input! (If you need any help let me know.)